tests/unit/plugins/module_utils/test_attachment.py

```diff
@@ -792,3 +794,42 @@ def test_changed_check_mode(self, client, tmp_path):
                 "file_name": "attachment_name.txt",
             },
         ] == sorted(record, key=lambda k: k["file_name"])
+
+
+class TestAttachmentGetAttachment:
+    def test_get_attachment(self, client):
+        client.get.return_value = Response(
+            200,
+            to_bytes("binary_data"),
+            {"headers": "headers"},
+        )
+        a = attachment.AttachmentClient(client)
+
+        response = a.get_attachment("0061f0c510247200964f77ffeec6c4de")
+
+        client.get.assert_called_once_with(
+            "api/now/attachment/0061f0c510247200964f77ffeec6c4de/file"
+        )
+        assert response.status == 200
+        assert response.data == to_bytes("binary_data")
+        assert response.headers == {"headers": "headers"}
+
+
+class TestAttachmentSaveAttachment:
+    def test_save_attachment(self, client, tmp_path):
+        path = tmp_path / "test.txt"
+        a = attachment.AttachmentClient(client)
+
+        a.save_attachment(to_bytes("test"), path)
+        file = open(str(path), "r")
+        b_data = file.read()
+
+        assert to_text(b_data) == "test"
+
+    def test_save_attachment_bad_dest(self, client):
+        a = attachment.AttachmentClient(client)
+
+        with pytest.raises(errors.ServiceNowError) as exc:
+            a.save_attachment(to_bytes("test"), "/not/a/path")
```

Comment:
If the directory on path `/not/a/` actually exists, the test will fail.

Reply:
Thank you for the comment! As suggested, tmpdir was used to generate nonexisting path